schema/core.go

```diff
@@ -4,13 +4,26 @@ import (
 	"encoding/json"
 	"fmt"
 	"regexp"
+	"strconv"
 	"strings"
 
 	datetime "github.com/di-wu/xsd-datetime"
 	"github.com/elimity-com/scim/errors"
 	"github.com/elimity-com/scim/optional"
 )
 
+var (
+	schemaAllowStringValues = false
+)
+
+// SetAllowStringValues sets whether string values are allowed.
+// If enabled, string values are allowed for booleans, integer and decimal attributes.
+// NOTE: This is NOT a standard SCIM behaviour, and should only be used for compatibility with non-compliant SCIM
+// clients, such as the one provided by Microsoft Azure.
+func SetAllowStringValues(enabled bool) {
```

Comment:
To make the behavior more granular, shouldn't it be an option for each boolean attribute instead of a global flag?

Reply:
I want to refrain from adding non-compliant interfaces structures in the library. When you define an attribute the SCIM protocol does not offer you this option. In general I am already reluctant to add this feature.

Reply:
Got it. Yes, unfortunately, this is something that is not compliant but met in the real world. 
This change will help eliminate some headaches for lots of other devs, including me, so I'm grateful for it :D